monkey/monkey_island/cc/repositories/agent_configuration_validation_decorator.py

```diff
@@ -0,0 +1,52 @@
+from jsonschema import validate
+
+from common.agent_configuration import AgentConfiguration
+from monkey_island.cc.repositories import (
+    IAgentConfigurationRepository,
+    PluginConfigurationValidationError,
+    RetrievalError,
+)
+from monkey_island.cc.repositories.utils import AgentConfigurationSchemaCompiler
+
+
+class AgentConfigurationValidationDecorator(IAgentConfigurationRepository):
+    """
+    A IAgentConfigurationRepository decorator that validates the agent configuration
+    """
+
+    def __init__(
+        self,
+        agent_configuration_repository: IAgentConfigurationRepository,
+        agent_configuration_schema_compiler: AgentConfigurationSchemaCompiler,
+    ):
+        self._agent_configuration_repository = agent_configuration_repository
+        self._agent_configuration_schema_compiler = agent_configuration_schema_compiler
```

Comment:
Can't we just use the schema?

Reply:
Do you mean `AgentConfiguration.schema()`? I don't think so, as the compiler is changing that schema with the plugins.